systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/BasicExternalKafkaClient.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.kafkaclients.externalClients;
+
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.kafkaclients.AbstractKafkaClient;
+import io.strimzi.systemtest.kafkaclients.IKafkaClientOperations;
+import io.strimzi.systemtest.kafkaclients.KafkaClientProperties;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.vertx.core.Vertx;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.IntPredicate;
+
+/**
+ * The BasicExternalKafkaClient for sending and receiving messages with basic properties. The client is using an external listeners.
+ */
+public class BasicExternalKafkaClient extends AbstractKafkaClient implements AutoCloseable, IKafkaClientOperations<Future<Integer>> {
+
+    private static final Logger LOGGER = LogManager.getLogger(BasicExternalKafkaClient.class);
+    private Vertx vertx = Vertx.vertx();
+
+    public static class Builder extends AbstractKafkaClient.Builder<Builder> {
+
+        @Override
+        public BasicExternalKafkaClient build() {
+            return new BasicExternalKafkaClient(this);
+        }
+
+        @Override
+        protected Builder self() {
+            return this;
+        }
+    }
+
+    private BasicExternalKafkaClient(Builder builder) {
+        super(builder);
+    }
+
+    public Future<Integer> sendMessagesPlain() {
+        return sendMessagesPlain(Constants.GLOBAL_CLIENTS_TIMEOUT);
+    }
+
+    /**
+     * Send messages to external entrypoint of the cluster with PLAINTEXT security protocol setting
+     * @return future with sent message count
+     */
+    @Override
+    public Future<Integer> sendMessagesPlain(long timeoutMs) {
+
+        String clientName = "sender-plain-" + this.clusterName;
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        vertx.deployVerticle(new Producer(
+            new KafkaClientProperties.KafkaClientPropertiesBuilder()
+                .withNamespaceName(namespaceName)
+                .withClusterName(clusterName)
+                .withBootstrapServerConfig(getExternalBootstrapConnect(namespaceName, clusterName))
+                .withKeySerializerConfig(StringSerializer.class.getName())
```

Comment:
Why does the `withKeySerializerConfig()` class take a `String` if it could take a `Class`?

Reply:
Yeah, I can do it to use Class instead of passing Strings thanks!